apps/compounds_tester/lib/compounds_tester_web/components/compounds/icon.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Compounds.Icon do
+  use Phoenix.Component
+
+  @moduledoc """
+   A simple icon component. It could be used to render an SVG OR an image icon.
+
+  # Examples:
+
+  ## Image
+  Don't provide a block to this component if you are using the src attribute.
+  ```html
+  <Compounds.Icon.icon src="/images/logo.svg"/>
+  ```
+  NOT
+  ```html
+    <Compounds.Icon.icon src="/images/logo.svg">
+    </Compounds.Icon.icon>
+  ```
+
+  ## SVG
+  Simply provide an SVG element as a child to this component.
+  DON'T provide a `src` attribute if you are using an SVG element.
+  ```html
+    <Compounds.Icon.icon>
+    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="currentColor" class="w-6 h-6">
+    <path d="M11.7 2.805a.75.75 0 01.6 0A60.65 60.65 0 0122.83 8.72a.75.75 0 01-.231 1.337 49.949 49.949 0 00-9.902 3.912l-.003.002-.34.18a.75.75 0 01-.707 0A50.009 50.009 0 007.5 12.174v-.224c0-.131.067-.248.172-.311a54.614 54.614 0 014.653-2.52.75.75 0 00-.65-1.352 56.129 56.129 0 00-4.78 2.589 1.858 1.858 0 00-.859 1.228 49.803 49.803 0 00-4.634-1.527.75.75 0 01-.231-1.337A60.653 60.653 0 0111.7 2.805z" />
+    <path d="M13.06 15.473a48.45 48.45 0 017.666-3.282c.134 1.414.22 2.843.255 4.285a.75.75 0 01-.46.71 47.878 47.878 0 00-8.105 4.342.75.75 0 01-.832 0 47.877 47.877 0 00-8.104-4.342.75.75 0 01-.461-.71c.035-1.442.121-2.87.255-4.286A48.4 48.4 0 016 13.18v1.27a1.5 1.5 0 00-.14 2.508c-.09.38-.222.753-.397 1.11.452.213.901.434 1.346.661a6.729 6.729 0 00.551-1.608 1.5 1.5 0 00.14-2.67v-.645a48.549 48.549 0 013.44 1.668 2.25 2.25 0 002.12 0z" />
+    <path d="M4.462 19.462c.42-.419.753-.89 1-1.394.453.213.902.434 1.347.661a6.743 6.743 0 01-1.286 1.794.75.75 0 11-1.06-1.06z" />
+    </svg>
+  </Compounds.Icon.icon>
+  ```
+  """
+
+  slot :inner_block
+  attr :src, :string
+  attr :class, :string, default: nil
+
+  def icon(assigns) do
+    assigns = assign(assigns, :src_exists, assigns[:src] != nil)
+    assigns = assign(assigns, :inner_block_exists, not Enum.empty?(assigns.inner_block))
+
+    if xnor(assigns.src_exists, assigns.inner_block_exists) do
+      raise "You must provide EITHER an SVG element as a child to this component,
```

Comment:
Should we handle this as a warning log? It begs the question if anything "breaking" in the components should handle it gracefully and log a warning or fail compilation.

Reply:
Hmm, yeah, I thought about it and it definitely shouldn't just break. I'll default it to the `src` and log some warnings if both are passed in.